src/Operations.jl

```diff
@@ -662,6 +662,8 @@ function download_source(ctx::Context; readonly=true)
         push!(pkgs_to_install, (;pkg, urls, path))
     end
 
+    OFFLINE_MODE[] && return Set{UUID}(entry.pkg.uuid for entry in pkgs_to_install)
```

Comment:
I think if we get here with non-empty `pkgs_to_install` in `offline` mode that seems like a bug so this feels like it is kind of just painting over the real issue.